packages/playwright-core/src/server/injected/recorder/recorder.ts

```diff
@@ -259,6 +259,25 @@ class RecordActionTool implements RecorderTool {
     });
   }
 
+  onContextMenu(event: MouseEvent) {
+    if (this._shouldIgnoreMouseEvent(event))
+      return;
+    if (this._actionInProgress(event))
+      return;
+    if (this._consumedDueToNoModel(event, this._hoveredModel))
+      return;
+
+    this._performAction({
+      name: 'click',
+      selector: this._hoveredModel!.selector,
+      position: positionForEvent(event),
+      signals: [],
+      button: 'right',
+      modifiers: 0,
+      clickCount: event.detail
```

Comment:
I believe `event.detail` is 0 for `contextmenu` event and we should set it to 0 here.

Reply:
https://github.com/microsoft/playwright/pull/31634/commits/6a8e68c656aa0f0e687effe0829ee74d4c99b41d